webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import Disk from './disk';
+import Checkbox from '../../../../common/forms/Checkbox';
+import './controller.scss';
+
+const Controller = ({
+  addDiskEnabled,
+  addDisk,
+  removeDisk,
+  updateController,
+  updateDisk,
+  ControllerTypes,
+  controller
+}) => {
+  const _updateController = (attribute, e) => {
+    const value = e.target.type === 'checkbox' ? e.target.checked : e.target.value;
+
+    updateController({[attribute]: value});
+  };
+
+  const _updateDisk = (diskIndex, attribute, e) => {
```

Comment:
they each dispatch a different action - 
one handles changes for the controller object,
the other for disk object.